Arashayo/TabletopHaven
# Remake rating system

## Related Issue
https://app.clickup.com/t/86bxfvgqq

## Changes list

- Remade rating system (relation between user and game). With old rating system it was impossible to make reliable rating system.
- Made system so users can rate games

## Testing instructions
Run `node backend/src/createTables.js`
Run `node backend/src/fillGames.js`
Go to a game and try rating it while logged in -> Expected output: Rating successfull (additionaly check if rating dynamically updates in the view)
Go to a game and try raing it while looged out -> Expected output: Error: "User needs to be logged in!" (check if rating does not change)
Check if you can see your games' rating after rating the game and refreshing the page -> Expected output: Rating persists updated when logged in, rating persists not updated when not logged in
Check if you can delete your games' rating (by clicking on trash icon) -> Expected output: Rating removed successfully
Check if trash icon is visible if you are not logged in -> Expected output: Trash icon not visible
Check if games' rating updates on other views after you rate it -> Expected output: Rating updates everywhere when the game occurs in the application

## Screenshots (if appropriate):
